muikku-core-plugins/src/main/resources/META-INF/resources/scripts/gui/communicator.js

```diff
@@ -206,41 +203,54 @@ $(document).ready(function(){
       }
     },    
     
-    _deleteMessages : function(ids) {
+    _deleteMessages : function(ids){
+      
+      
       var _this = this;
-      for (i = 0; i < ids.length; i++) {
-        var endpoint = mApi().communicator.items;
-        var hash = window.location.hash.substring(1);
-        
+      var messages = ids.length;
+      var endpoint = mApi().communicator.items;
+      var hash = window.location.hash.substring(1);
+
+      var loadNotification = $('.notification-queue').notificationQueue('notification', 'loading', getLocaleText('plugin.communicator.infomessage.delete.deleting', messages));
+
         if (hash == "sent") {
           endpoint = mApi().communicator.sentitems;
         }
-
-        endpoint.del(ids[i]).callback(function (err, result){
-         if (err) {
-            $('.notification-queue').notificationQueue('notification', 'error', getLocaleText('plugin.communicator.infomessage.delete.error'));
-          } else {
-            $('.notification-queue').notificationQueue('notification', 'success', getLocaleText('plugin.communicator.infomessage.delete.success'));
-          }         
-          _this._refreshView();
-        });
-      } 
+      
+      
+      var batch = $.map(ids, function(id){
+        endpoint.del(id);
```

Comment:
This yields to an empty list and thus the call is ignored completely, fix by changing to return endpoint.del(id);